op-node/rollup/derive/l1_retrieval.go

```diff
@@ -2,37 +2,24 @@ package derive
 
 import (
 	"context"
-	"fmt"
 	"io"
 
 	"github.com/ethereum-optimism/optimism/op-node/eth"
 	"github.com/ethereum/go-ethereum/log"
 )
 
-// This is a generic wrapper around fetching all transactions in a block & then
-// it feeds one L1 transaction at a time to the next stage
-
-// DataIter is a minimal iteration interface to fetch rollup input data from an arbitrary data-availability source
-type DataIter interface {
-	// Next can be repeatedly called for more data, until it returns an io.EOF error.
-	// It never returns io.EOF and data at the same time.
-	Next(ctx context.Context) (eth.Data, error)
-}
-
-// DataAvailabilitySource provides rollup input data
-type DataAvailabilitySource interface {
-	// OpenData does any initial data-fetching work and returns an iterator to fetch data with.
-	OpenData(ctx context.Context, id eth.BlockID) (DataIter, error)
-}
-
 type L1SourceOutput interface {
 	StageProgress
 	IngestData(data []byte)
 }
 
+type CalldataProvider interface {
+	OpenData(ctx context.Context, id eth.BlockID) DataIter
+}
+
 type L1Retrieval struct {
 	log     log.Logger
-	dataSrc DataAvailabilitySource
+	dataSrc CalldataProvider
 	next    L1SourceOutput
```

Comment:
Can we keep the name more general here? It's meant to be swapped out for EIP-4844 and other data sources easily, which are not calldata.